osgp/platform/osgp-core/src/test/java/org/opensmartgridplatform/core/application/services/EventNotificationMessageServiceTest.java

```diff
@@ -0,0 +1,71 @@
+/**
+ * Copyright 2020 Alliander N.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+package org.opensmartgridplatform.core.application.services;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+import org.joda.time.DateTime;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.opensmartgridplatform.core.domain.model.domain.DomainRequestService;
+import org.opensmartgridplatform.domain.core.entities.Event;
+import org.opensmartgridplatform.domain.core.exceptions.UnknownEntityException;
+import org.opensmartgridplatform.domain.core.valueobjects.EventType;
+import org.opensmartgridplatform.dto.valueobjects.EventNotificationDto;
+import org.opensmartgridplatform.dto.valueobjects.EventTypeDto;
+import org.opensmartgridplatform.shared.domain.services.CorrelationIdProviderTimestampService;
+import org.opensmartgridplatform.shared.infra.jms.MessageType;
+
+@ExtendWith(MockitoExtension.class)
+public class EventNotificationMessageServiceTest {
+
+    @Mock
+    private CorrelationIdProviderTimestampService correlationIdProviderTimestampService;
+
+    @Mock
+    private DomainRequestService domainRequestService;
+
+    @Mock
+    private EventNotificationHelperService eventNotificationHelperService;
+
+    @InjectMocks
+    private EventNotificationMessageService eventNotificationMessageService;
+
+    @Test
+    void handleLightSensorReportsLightEventTest() throws UnknownEntityException {
+        final String deviceUid = "testUid";
+        final String deviceIdentification = "testIdentification";
+        final DateTime dateTime = DateTime.now();
+        final EventTypeDto eventTypeDto = EventTypeDto.LIGHT_SENSOR_REPORTS_LIGHT;
+        final String description = "Sensor reports light";
+        final Integer index = 0;
+        final EventNotificationDto eventNotificationDto = new EventNotificationDto(deviceUid, dateTime, eventTypeDto,
+                description, index);
+
+        final EventNotificationMessageService eventNotificationMessageServiceSpy = spy(
+                this.eventNotificationMessageService);
+        eventNotificationMessageServiceSpy.handleEvent(deviceIdentification, eventNotificationDto);
```

Comment:
The handleEvent(String, EventNotificationDto) method calls handleEvent(String, Date, ...). Directly call that second method. You could write a separate test if you want to show handleEvent(String, EventNotificationDto) calls handleEvent(String, Date, ...) correctly.

Reply:
But this tests results in a bit higher code coverage ;-)